src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-cisco.dfdl.xsd

```diff
@@ -0,0 +1,136 @@
+      <xs:documentation>
+        Defines a single syslog message of the Cisco format, including the
+        facility and severity. This should be used as the root element if the
+        goal is to only support Cisco messages.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:complexType>
+      <xs:sequence>
+        <xs:group ref="sl:Prival" />
+        <xs:group ref="sl:Cisco" />
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:complexType name="Cisco">
+    <xs:annotation>
+      <xs:documentation>
+        Defines a type containing the content of a Cisco syslog message,
+        excluding the facility and severity. This should be used when the
+        faciliy/severity are already parsed.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:group ref="sl:Cisco" />
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:group name="Cisco">
+    <xs:annotation>
+      <xs:documentation>
+        Defines all content of a Cisco syslog message that appears immediately
+        affter the severify/facility part of a syslog message.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element name="Header">
+        <xs:complexType>
+          <xs:sequence dfdl:separator="%SP;" dfdl:separatorPosition="postfix">
+            <xs:element name="SequenceNumber" type="xs:int" dfdl:textNumberPattern="0" dfdl:terminator=":" />
+            <xs:element name="TimestampMonth" dfdl:lengthKind="explicit" dfdl:length="3">
+              <xs:simpleType>
+                <xs:restriction base="xs:string">
+                  <xs:enumeration value="Jan" />
+                  <xs:enumeration value="Feb" />
+                  <xs:enumeration value="Mar" />
+                  <xs:enumeration value="Apr" />
+                  <xs:enumeration value="May" />
+                  <xs:enumeration value="Jun" />
+                  <xs:enumeration value="Jul" />
+                  <xs:enumeration value="Aug" />
+                  <xs:enumeration value="Sep" />
+                  <xs:enumeration value="Oct" />
+                  <xs:enumeration value="Nov" />
+                  <xs:enumeration value="Dec" />
+                </xs:restriction>
+              </xs:simpleType>
+            </xs:element>
+            <xs:element name="TimestampDay" dfdl:lengthKind="explicit" dfdl:length="2" dfdl:textNumberPattern="* 00">
+              <xs:simpleType>
+                <xs:restriction base="xs:int">
+                  <xs:minInclusive value="1" />
+                  <xs:maxInclusive value="31" />
+                </xs:restriction>
+              </xs:simpleType>
+            </xs:element>
+            <xs:element name="TimestampTime" type="xs:time" dfdl:calendarPattern="HH:mm:ss" dfdl:lengthKind="explicit" dfdl:length="8" dfdl:terminator=":" />
+            <xs:sequence dfdl:initiator="%%" dfdl:separator="-" dfdl:terminator=":">
+              <xs:element name="Facility" type="xs:string" />
```

Comment:
Unconstrained string type will fail XSAT-type quality analysis.  Another reason to define shared types is that basically you can never use xs:string except as a base in a simpleType def, if you want to keep XSAT-style checkers happy.